website/docs: improve-rac-documents (#14414)
goauthentik/authentik: website/docs/add-secure-apps/providers/rac/rac-public-key.md

---
title: RAC SSH Public Key Authentication
---

## About RAC SSH public key authentication

> The RAC provider supports SSH public key authentication. This allows for secure connections to SSH endpoints without the use of username and passwords.

Reviewer: You'd still need to specify the user you're connecting to

> SSH private keys can be set via several methods:

Reviewer: ```suggestion
SSH private keys can be configured using various methods:
```

Reviewer: IMO

1. Log in to authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Applications** > **Providers**.
3. Click the **Edit** icon on the RAC provider that requires public key authentication.
4. In the **Settings** codebox enter the private key of the endpoint, for example:
    ```python
    private-key:
        -----BEGIN SSH PRIVATE KEY-----
        SAMPLEgIBAAJBAKj34GkxFhD90vcNLYLInFEX6Ppy1tPf9Cnzj4p4WGeKLs1Pt8Qu
        KUpRKfFLfRYC9AIKjbJTWit+CqvjWYzvQwECAwEAAQJAIJLixBy2qpFoS4DSmoEm
        o3qGy0t6z09AIJtH+5OeRV1be+N4cDYJKffGzDa88vQENZiRm0GRq6a+HPGQMd2k
        TQIhAKMSvzIBnni7ot/OSie2TmJLY4SwTQAevXysE2RbFDYdAiEBCUEaRQnMnbp7
        9mxDXDf6AU0cN/RPBjb9qSHDcWZHGzUCIG2Es59z8ugGrDY+pxLQnwfotadxd+Uy
        v/Ow5T0q5gIJAiEAyS4RaI9YG8EWx/2w0T67ZUVAw8eOMB6BIUg0Xcu+3okCIBOs
        /5OiPgoTdSy7bcF9IGpSE8ZgGKzgYQVZeN97YE00
        -----END SSH PRIVATE KEY-----
    ```
5. Click **Update**.

## How to apply a private key to an RAC endpoint

1. Log in to authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Applications** > **Providers**.
3. Click the name of the RAC provider that the endpoint belongs to.

> 4. Under **Endpoints**- click on the **Edit** icon next to the endpoint that requires public key authentication.

Reviewer: ```suggestion
4. Under **Endpoints** click on the **Edit** icon next to the endpoint that requires public key authentication.
```

> 5. Under **Advanced settings**, in the **Settings** codebox enter the private key of the endpoint, for example:

Reviewer: I wouldn't use "for example" as that is the config one needs to use. Only the SSH key is configurable, i'd update the wording

    ```python
    private-key:
        -----BEGIN SSH PRIVATE KEY-----
        SAMPLEgIBAAJBAKj34GkxFhD90vcNLYLInFEX6Ppy1tPf9Cnzj4p4WGeKLs1Pt8Qu
        KUpRKfFLfRYC9AIKjbJTWit+CqvjWYzvQwECAwEAAQJAIJLixBy2qpFoS4DSmoEm
        o3qGy0t6z09AIJtH+5OeRV1be+N4cDYJKffGzDa88vQENZiRm0GRq6a+HPGQMd2k
        TQIhAKMSvzIBnni7ot/OSie2TmJLY4SwTQAevXysE2RbFDYdAiEBCUEaRQnMnbp7
        9mxDXDf6AU0cN/RPBjb9qSHDcWZHGzUCIG2Es59z8ugGrDY+pxLQnwfotadxd+Uy
        v/Ow5T0q5gIJAiEAyS4RaI9YG8EWx/2w0T67ZUVAw8eOMB6BIUg0Xcu+3okCIBOs
        /5OiPgoTdSy7bcF9IGpSE8ZgGKzgYQVZeN97YE00
        -----END SSH PRIVATE KEY-----
    ```
5. Click **Update**.

## How to apply a private key to an RAC property mapping

1. Log in to authentik as an admin, and open the authentik Admin interface.

> 2. Navigate to **Customization** > **Property Mappings** and click **Create**. Create a **RAC Provider Property Mapping** with the following settings:

Reviewer: ```suggestion
2. Navigate to **Customization** > **Property Mappings** and click **Create**, then create a **RAC Provider Property Mapping** with the following settings:
```

    - **Name**: Choose a descriptive name

>     - Under **Advanced Settings** in the **Expression** codebox:

Reviewer: ```suggestion
    - In the **Expression** codebox under **Advanced Settings**:
```

Reviewer: TBH I'm not a fan of either wording. Do whatever you think is best

Author: I've changed it to match some integration docs - I think that it works better this way. What do you think?

        o3qGy0t6z09AIJtH+5OeRV1be+N4cDYJKffGzDa88vQENZiRm0GRq6a+HPGQMd2k
        TQIhAKMSvzIBnni7ot/OSie2TmJLY4SwTQAevXysE2RbFDYdAiEBCUEaRQnMnbp7
        9mxDXDf6AU0cN/RPBjb9qSHDcWZHGzUCIG2Es59z8ugGrDY+pxLQnwfotadxd+Uy
        v/Ow5T0q5gIJAiEAyS4RaI9YG8EWx/2w0T67ZUVAw8eOMB6BIUg0Xcu+3okCIBOs
        /5OiPgoTdSy7bcF9IGpSE8ZgGKzgYQVZeN97YE00
        -----END SSH PRIVATE KEY-----",
        }
        ```
3. Click **Finish**.
4. Navigate to **Applications** > **Providers**.
5. Click the **Edit** icon on the RAC provider that requires public key authentication.
6. Under **Protocol Settings** add the newly created property mapping to **Selected Property Mappings**.
7. Click **Update**.

## How to fetch a private key from a user's attributes and apply it to an RAC property mapping

1. Log in to authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Customization** > **Property Mappings** and click **Create**. Create a **RAC Provider Property Mapping** with the following settings:

    - **Name**: Choose a descriptive name
    - Under **Advanced Settings** in the **Expression** codebox:
        ```python
        return {

>         "private-key": request.user.attributes.get("private-key-attribute-name", "default"),

Reviewer: ```suggestion
        "private-key": request.user.attributes.get("<private-key-attribute-name>", "default"),
```

Since it's a placeholder

        }
        ```
3. Click **Finish**.
4. Navigate to **Applications** > **Providers**.
5. Click the **Edit** icon on the RAC provider that requires public key authentication.

> 6. Under **Protocol Settings** add the newly created property mapping to **Selected Property Mappings**.

Reviewer: ```suggestion
6. Under **Protocol Settings**, add the newly created property mapping to **Selected Property Mappings**.
```

7. Click **Update**.

:::note

> For group attributes the following expression can be used `request.user.group_attributes(request.http_request)`

Reviewer: ```suggestion
For group attributes, the following expression can be used: `request.user.group_attributes(request.http_request)`
```

---
title: RAC SSH Public Key Authentication
---

## About RAC SSH public key authentication

The RAC provider supports SSH public key authentication. This allows for secure connections to SSH endpoints without the use of passwords.

> SSH private keys can be configured via several methods:

Reviewer 2: Is this the same info as in file above, on line 57 ("RAC connection settings can be set via several methods:" ...?

Ideally we would not repeat the same exact chunk of content, or if we absolutely need to then we cold choose (if worth the effort) to use a "snippet" that lives in one place (probably [here](https://github.com/goauthentik/authentik/tree/main/website/docs/expressions) and gets used wherever we want.

---
title: RAC SSH Public Key Authentication
---

## About RAC SSH public key authentication

The RAC provider supports SSH public key authentication. This allows for secure connections to SSH endpoints without the use of username and passwords.

## Current limitations

> Each RAC provider is limited to a single private key. This means that a RAC provider with a private key set, will utilize that private key for all connections.

Reviewer 3: That's incorrect, the private key can also be set on a property mapping and in the endpoint settings and can be fetched from user attributes
